### ISSUE TITLE
test dataset

### ISSUE THREAD

**Issue by mompiou** [YBZh/Bridging_UDA_SSL#2]
Thank you for this nice work. I was trying to apply SSL (Mean Teacher) for DA with my own dataset. I have synthetic data labelled (source), unlabelled data (target) and some labeled data from the target.  Am I correct to use part of source target for validation set for weights update ? How to use labelled data from target ? What is the trans_test dataset in your code ?

**Comment by YBZh** [comment]
Am I correct to use part of source target for validation set for weights update ?  --> I am not certain since I do not fully understand your question.

How to use labelled data from target ? --> The current code base seems not support the use of labeled target data. You need to implement it by yourself.

What is the trans_test dataset in your code ? --> In the transductive transfer learning setting, the unlabeled test data are used in the training process. Therefore, the trans_test is exactly the actual test data. In the inductive transfer learning setting, the trans_test and the actual test data are two different sampling from the same target domain.

**Comment by mompiou** [comment]
My main concern is how to deal with target data that are actually already labelled. As target data are unlabelled here I don't know how to use them (contrary to SSL where I use them in the supervised branch).

**Comment by YBZh** [comment]
Ok, I see. The current code base does not support three streams of input: labeled source data, labeled source data, and unlabeled source data. My suggestion is to add another "dataset" and corresponding "dataloader" in the [prepare_data](https://github.com/YBZh/Bridging_UDA_SSL/blob/76bbe39b1bd3f52ab8053cc09fa378b5be2fe4c9/data/prepare_data_da.py)
